core/src/test/java/kafka/test/junit/RaftClusterInvocationContext.java

```diff
@@ -86,67 +81,42 @@ public String getDisplayName(int invocationIndex) {
 
     @Override
     public List<Extension> getAdditionalExtensions() {
-        RaftClusterInstance clusterInstance = new RaftClusterInstance(clusterReference, zkReference, clusterConfig, isCombined);
+        RaftClusterInstance clusterInstance = new RaftClusterInstance(clusterConfig, isCombined);
         return Arrays.asList(
-            (BeforeTestExecutionCallback) context -> {
-                TestKitNodes nodes = new TestKitNodes.Builder().
-                        setBootstrapMetadataVersion(clusterConfig.metadataVersion()).
-                        setCombined(isCombined).
-                        setNumBrokerNodes(clusterConfig.numBrokers()).
-                        setPerServerProperties(clusterConfig.perServerOverrideProperties()).
-                        setNumDisksPerBroker(clusterConfig.numDisksPerBroker()).
-                        setNumControllerNodes(clusterConfig.numControllers()).build();
-                KafkaClusterTestKit.Builder builder = new KafkaClusterTestKit.Builder(nodes);
-
-                if (Boolean.parseBoolean(clusterConfig.serverProperties().getOrDefault("zookeeper.metadata.migration.enable", "false"))) {
-                    zkReference.set(new EmbeddedZookeeper());
-                    builder.setConfigProp("zookeeper.connect", String.format("localhost:%d", zkReference.get().port()));
-                }
-                // Copy properties into the TestKit builder
-                clusterConfig.serverProperties().forEach(builder::setConfigProp);
-                // KAFKA-12512 need to pass security protocol and listener name here
-                KafkaClusterTestKit cluster = builder.build();
-                clusterReference.set(cluster);
-                cluster.format();
-                if (clusterConfig.isAutoStart()) {
-                    cluster.startup();
-                    kafka.utils.TestUtils.waitUntilTrue(
-                        () -> cluster.brokers().get(0).brokerState() == BrokerState.RUNNING,
-                        () -> "Broker never made it to RUNNING state.",
-                        org.apache.kafka.test.TestUtils.DEFAULT_MAX_WAIT_MS,
-                        100L);
-                }
-            },
-            (AfterTestExecutionCallback) context -> clusterInstance.stop(),
-            new ClusterInstanceParameterResolver(clusterInstance)
+                (BeforeTestExecutionCallback) context -> {
+                    if (clusterConfig.isAutoStart()) {
```

Comment:
In order to keep compatibility (less changes to tests), we should add `clusterInstance.format();`